4337/package.json

```diff
@@ -43,46 +43,37 @@
     "url": "https://github.com/safe-global/safe-modules/issues"
   },
   "devDependencies": {
-    "@nomiclabs/hardhat-ethers": "^2.0.6",
-    "@nomiclabs/hardhat-etherscan": "^3.1.0",
-    "@nomiclabs/hardhat-waffle": "^2.0.3",
+    "@nomicfoundation/hardhat-toolbox": "^3.0.0",
     "@openzeppelin/contracts": "^5.0.0",
     "@types/chai": "^4.3.1",
     "@types/mocha": "^9.1.1",
-    "@types/node": "^17.0.42",
+    "@types/node": "^18.18.6",
     "@types/yargs": "^17.0.10",
     "@typescript-eslint/eslint-plugin": "^5.28.0",
     "@typescript-eslint/parser": "^5.28.0",
-    "argv": "^0.0.2",
-    "chai": "^4.2.0",
     "debug": "^4.3.4",
     "dotenv": "^16.0.1",
     "eslint": "^8.17.0",
     "eslint-config-prettier": "^8.5.0",
     "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-no-only-tests": "^2.4.0",
     "eslint-plugin-prettier": "^4.0.0",
-    "ethereum-waffle": "^3.4.4",
-    "ethers": "^5.6.8",
-    "hardhat": "2.9.9",
-    "hardhat-deploy": "0.11.10",
+    "ethers": "^6.8.0",
+    "hardhat": "2.18.2",
+    "hardhat-deploy": "0.11.43",
     "husky": "^8.0.1",
     "prettier": "^2.7.0",
     "prettier-plugin-solidity": "^1.0.0-beta.19",
-    "solc": "0.8.14-fixed",
+    "solc": "0.8.21",
     "solhint": "^3.3.7",
     "solhint-plugin-prettier": "^0.0.5",
-    "solidity-coverage": "^0.7.21",
     "ts-node": "^10.8.1",
     "typescript": "^4.7.3",
     "yargs": "^17.5.1"
   },
-  "peerDependencies": {
-    "ethers": "^5.1.4"
-  },
   "overrides": {
     "@safe-global/safe-contracts": {
-      "ethers": "^5.6.8"
+      "ethers": "^6.8.0"
```

Comment:
> Or does npm not allow installing dependencies because of incompatible packages in case there's no override?

The issue I ran into is that both `@nomicfoundation/hardhat-toolbox` and `@safe-global/safe-contracts` have an `ethers` peer dependency with non-overlapping version ranges. AFAICT, NPM does not support this (looked for a bit online but didn't find anything conclusive, I may be wrong though).

Furthermore, I noticed that we use `@safe-global/safe-contracts` only for contracts and not exported TypeScript/JavaScript code, so even if we override the `ethers` version, it should not cause any problems.

And finally, the other thing that I found confusing was that [the `package.json` in the `safe-contracts` repo](https://github.com/safe-global/safe-contracts/blob/main/package.json) does not include an `ethers` dependency or peer dependency, while the one [published to NPM does](https://unpkg.com/browse/@safe-global/safe-contracts@1.4.1/package.json).

Reply:
> And finally, the other thing that I found confusing was that [the package.json in the safe-contracts repo](https://github.com/safe-global/safe-contracts/blob/main/package.json) does not include an ethers dependency or peer dependency, while the one [published to NPM does](https://unpkg.com/browse/@safe-global/safe-contracts@1.4.1/package.json).

I think it was a result of the upgrade to the hardhat-toolbox as well since ethers is already a dependency of the hardhat-toolbox package. But this upgrade happened after 1.4.1, which has not been released yet.